resharper/resharper-unity/src/Core/Psi/Modules/UnityExternalFilesModuleProcessor.cs

```diff
@@ -179,45 +145,98 @@ private void CollectExternalFilesForDirectory(ExternalFiles externalFiles, Virtu
             // Mac, on the same project!
             var entries = directory.GetDirectoryEntries("*", PathSearchFlags.RecurseIntoSubdirectories
                                                              | PathSearchFlags.ExcludeDirectories);
+
             foreach (var entry in entries)
-                externalFiles.AddFile(entry);
+            {
+                // Ignore anything under a folder that ends with a tilde - Unity does not import these folders into the
+                // asset database, so they're not part of this project
+                if (entry.IsFile && !entry.RelativePath.FullPath.Contains("~/") &&
```

Comment:
What do you think about using GetDirectoryEntries without recursive mode and manually enter directories which name does not contain `~`? In that case we will scan shorter lines + we will not scan child files inside directory with `~` at all

Reply:
Hmm. The majority of files will be without the `~`, but it's probably worth unrolling it.

Reply:
Yes, the majority of files will be without `~` and for each file we will scan the whole full path string for `~` instead of short name of current file or directory, e.g let's consider several paths:
`PackageName/Assets/Folder1/Foo1/Foo2/Name1/`
...
`PackageName/Assets/Folder1/Foo1/Foo2/Name100/`

You will scan 100 full pathes for `~` instead of scanning `PackageName`, `Assets`, `Folder1`, `Foo1`, `Foo2` and names since `Name1` up to `Name100`. 105 short names instead of 100 full pathes. But possibly it will be fast.

Reply:
But I am not sure that unrolling GetDirectoryEntries will be same to GetDirectoryEntries with recursive flag, we could lost all benefits from unrolling if it is true